File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        if nums[-1] > nums[0]:
            return nums[0]
        low = 0
        high = len(nums) - 1
        ans = nums[-1]

        while (low <= high):
            mid = (low + high) // 2

            if (nums[mid] > ans):
                low = mid + 1
            elif (nums[mid] < ans):
                ans = nums[mid]
                high = mid - 1
            else:
                return ans
        return ans


        

        
        

File: Data Structures & Algorithms/counting-bits/submission-0.cpp
class Solution {
public:
    vector<int> countBits(int n) {

        vector<int> ans(n + 1);

        for (int i = 0; i <= n; i++) {
            int j = i;
            int cnt = 0;

            while (j) {
                j &= j - 1;
                cnt += 1;
            }

            ans[i] = cnt;

        }

        return ans;



        
    }
};
